packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -121,6 +121,377 @@ void main() {
+      expect(state.currentTextEditingValue.composing, equals(const TextRange(start: 4, end: 9)));
+      state.updateEditingValue(const TextEditingValue(text: 'hello الْعَ بِيَّةُ'));
+      state.updateEditingValue(const TextEditingValue(text: 'hello الْعَ بِيَّةُ '));
+      await tester.pump();
+      expect(state.renderEditable.text!.toPlainText(), equals('hello الْعَ بِيَّةُ \u{200F}'));
+    });
+
+    testWidgets("Whitespace directionality doesn't overwrite existing Arabic", (WidgetTester tester) async {
+      final TextEditingController controller = TextEditingController(text: 'testText');
+      await tester.pumpWidget(
+        MediaQuery(
+          data: const MediaQueryData(devicePixelRatio: 1.0),
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: FocusScope(
+              node: focusScopeNode,
+              autofocus: true,
+              child: EditableText(
+                backgroundCursorColor: Colors.blue,
+                controller: controller,
+                focusNode: focusNode,
+                maxLines: 1, // Sets text keyboard implicitly.
+                style: textStyle,
+                cursorColor: cursorColor,
+              ),
+            ),
+          ),
+        ),
+      );
+
+      await tester.tap(find.byType(EditableText));
+      await tester.showKeyboard(find.byType(EditableText));
+      controller.text = '';
+      await tester.idle();
+
+      final EditableTextState state =
+          tester.state<EditableTextState>(find.byType(EditableText));
+      expect(tester.testTextInput.editingState!['text'], equals(''));
+      expect(state.wantKeepAlive, true);
+
+      // Does not overwrite existing RLM or LRM characters
+      state.updateEditingValue(const TextEditingValue(text: 'hello \u{200F}ا'));
+      await tester.pump();
+      expect(state.textEditingValue.text, equals('hello \u{200F}ا'));
+      state.updateEditingValue(const TextEditingValue(text: 'hello \u{200F}ا \u{200E}ا ا '));
+      await tester.pump();
+      expect(state.textEditingValue.text, equals('hello \u{200F}ا \u{200E}ا ا '));
+
+      // Handles only directionality markers.
+      state.updateEditingValue(const TextEditingValue(text: '\u{200E}\u{200F}'));
+      await tester.pump();
+      expect(state.textEditingValue.text, equals('\u{200E}\u{200F}'));
+      state.updateEditingValue(const TextEditingValue(text: '\u{200E}\u{200F}\u{200E}\u{200F}\u{200E}\u{200F}'));
+      await tester.pump();
+      expect(state.textEditingValue.text, equals('\u{200E}\u{200F}\u{200E}\u{200F}\u{200E}\u{200F}'));
+      state.updateEditingValue(const TextEditingValue(text: '\u{200E}\u{200F}\u{200F}\u{200F}'));
+      await tester.pump();
+      expect(state.textEditingValue.text, equals('\u{200E}\u{200F}\u{200F}\u{200F}'));
+    });
+
+    testWidgets('Whitespace directionality is not leaky Arabic', (WidgetTester tester) async {
```

Comment:
The original test was trying to test the formatter doesn't delete "real" user input since it changes the text editing value.

Reply:
I think this test can probably be removed? /cc @GaryQian ?

Reply:
Cant hurt to leave it as it is relatively lightweight. It will ensure that repeatedly sending the same values doesn't result in changes. Not directly relevant, but I could see some crazy changes triggering this to fail.